File: Upcoder/Lập trình hướng đối tượng/1.Object and Class/Xây dựng lớp phân số/LOP1.2.PhanSo.cpp
#include<bits/stdc++.h>
using namespace std;
class Phanso
{
    private:
    int Tu;
    int Mau;
    int UCLN(int a, int b) {
    if (b == 0) 
        return a;
    return UCLN(b, a % b);
    }
    public:
    void Nhap();
    void Xuat();
    void Rutgon();
    void XuatTu();
    void XuatMau();
    void XuatNghich();
    void Tong();
};

int main()
{
    Phanso ps;
    ps.Nhap();
    ps.Xuat();
    ps.XuatTu();
    ps.XuatMau();
    ps.XuatNghich();
    ps.Rutgon();
    ps.Tong();
}
void Phanso::Nhap()
{
    cin >> Tu >> Mau;
}
void Phanso::Xuat()
{
    cout << Tu << "/" << Mau << endl;
}
void Phanso::XuatTu()
{
    cout << Tu << endl;
}

void Phanso::XuatMau()
{
    cout << Mau << endl;
}
void Phanso::XuatNghich()
{
    cout << Mau << "/" << Tu << endl;
}
void Phanso::Rutgon()
{
    int ucln = UCLN(Tu, Mau);
    Tu /= ucln;
    Mau /= ucln;
    cout << Tu << "/" << Mau << endl;
}
void Phanso::Tong()
{
    int a = Tu, b = Mau, c = Mau, d = Tu;
    int T = a * d + b * c;
    int M = b * d;
    int ucln = UCLN(T, M);
    T /= ucln;
    M /= ucln;
    cout << T << "/" << M;
}

File: Upcoder/Lập trình hướng đối tượng/1.Object and Class/Xây dựng lớp phân số/LOP1.2.PhanSo.cs
using System;
namespace PhanSo
{
    public class PhanSo
    {
        private int _tuSo;
        private int _mauSo;
        private int UCLN(int a, int b)
        {
            if(b == 0)
                return a;
            return UCLN(b, a % b);
        }
        public PhanSo()
        {
        }
        public void Nhap()
        {
            string[] input = Console.ReadLine().Split();
            _tuSo = int.Parse(input[0]);
            _mauSo = int.Parse(input[1]);
        }
        public void Xuat()
        {
            Console.WriteLine("{0}/{1}", _tuSo, _mauSo);
        }
        public void XuatNguoc()
        {
            Console.WriteLine("{0}/{1}", _mauSo, _tuSo);
        }
        public void Xuattu()
        {
            Console.WriteLine("{0}", _tuSo);
        }
        public void Xuatmau()
        {
            Console.WriteLine("{0}", _mauSo);
        }
        public void Rutgon()
        {
            int c = UCLN(_tuSo, _mauSo);
            _tuSo /= c;
            _mauSo /= c;
            Console.WriteLine("{0}/{1}", _tuSo, _mauSo);
        }
        public void Xuattong()
        {
            int a = _tuSo;
            int b = _mauSo;
            int c = _mauSo;
            int d = _tuSo;
            int T = a * d + b * c;
            int M = b * d;
            int ucln = UCLN(T, M);
            T /= ucln;
            M /= ucln;
            Console.Write("{0}/{1}", T, M);
        }
    }
    public class Program
    {
        static void Main(string[]args)
        {
            PhanSo ps = new PhanSo();
            ps.Nhap();
            ps.Xuat();
            ps.Xuattu();
            ps.Xuatmau();
            ps.XuatNguoc();
            ps.Rutgon();
            ps.Xuattong();
        }
    }
}